dozer/cogs/_utils.py

```diff
@@ -289,4 +299,56 @@ def decorator(func):
             func.__required_permissions__ = discord.Permissions()
             func.__required_permissions__.update(**required)
         return func
+
     return decorator
+
+
+class PrefixHandler:
+    """Handles dynamic prefixes"""
+
+    def __init__(self, default_prefix):
+        self.default_prefix = default_prefix
+        self.prefix_cache = {}
+
+    def handler(self, bot, message):
+        """Process the dynamic prefix for each message"""
+        dynamic = self.prefix_cache.get(message.guild.id if message.guild else 0)
+        return dynamic if dynamic else self.default_prefix
```

Comment:
One general ish comment - the bot status still uses the bot's default prefix for it, should we either implement some kind of fallback handling with the bot's mention? I feel like having the prefix in the user status can be counter productive in this instance.